lib/server-impl.js

```diff
@@ -26,22 +26,24 @@ function createApp(options) {
     );
 
     const app = getApp(config);
-    const server = app.listen(app.get('port'), () => {
-        logger.info(`Unleash started on ${app.get('port')}`);
-    });
-
     startMonitoring(options.serverMetrics, eventBus);
 
-    return { app, server, eventBus };
+    const server = app.listen({ port: options.port, host: options.host }, () =>
+        logger.info(`Unleash started on port ${server.address().port}`)
+    );
+
+    return new Promise((resolve, reject) => {
+        server.on('listening', () => resolve({ app, server, eventBus }));
+        server.on('error', reject);
+    });
 }
 
 function start(opts) {
     const options = createOptions(opts);
 
     return migrator({ databaseUrl: options.databaseUrl })
         .catch(err => logger.error('failed to migrate db', err))
```

Comment:
should we let this error bubble up? if migration fails, I don't want unleash to start up, do i?

Reply:
Agreed, will do it as a different issue though